openedx/core/djangoapps/content/learning_sequences/api/data.py

```diff
@@ -32,6 +32,8 @@
 from opaque_keys.edx.keys import CourseKey, UsageKey
 
 
+COURSE_VISIBILITIES = ["public", "public_outline", "private"]
```

Comment:
@ahmedbilal, can you use a enum instead?

Reply:
@lgp171188 I tried using it, but it cause issues with the `attr.validator.in_()` i.e It only accept element of that enum but not their value which we really want to save. For example,

```python
import enum

class CourseVisibilities(enum.Enum):
    PUBLIC = "public"
    PUBLIC_OUTLINE = "public_outline"
    PRIVATE = "private"
```

So, the `attr.validator.in_()` would accept `CourseVisibilities.PRIVATE` but not the value `private` which we really want to store. If we gave it `CourseVisibilities.PRIVATE`, it would pass the validator and get stored in the database in exactly the same way i.e in database the column would have value `CourseVisibilities.PRIVATE` instead of the value of that constant `private`

Reply:
I'm not clear on the concern here, but I'm also fuzzy on enums, so please bear with me. If we have a variable `course_visibility` with a value of `CourseVisibilities.PRIVATE` and we want to get the string `'private'`, we call `course_visibility.value` and get it, right?

In general, I would prefer that we use enums instead of strings when possible, since it makes the legal values clearer and it will fail earlier in the process if someone puts in bad data.

Reply:
Done